make/configure.py

```diff
@@ -1748,9 +1748,6 @@ class Tools:
     # Require FFmpeg AAC on Linux and Windows
     options.enable_ffmpeg_aac = IfHost(options.enable_ffmpeg_aac, '*-*-darwin*',
                                        none=True).value
-    # Allow GTK mingw only on mingw
-    options.enable_gtk_mingw  = IfHost(options.enable_gtk_mingw, '*-*-mingw*',
```

Comment:
I take it we can no longer cross compile GTK? We actually did use this in the past, but perhaps it's no longer relevant.

Reply:
It can still be done, but now it requires the `--enable-gtk` flag. Last I checked, it did build with some manual intervention, provided the necessary MinGW packages were installed.

Reply:
Unless something is broken/unfixable here, we shouldn't rip out support for cross-compiling the GTK GUI using mingw-w64. It's not popular but people use it, and in the past we've used it for development purposes.

Reply:
But the support is still in there? The only difference is that the flag has changed (and it applies to any operating system that doesn't enable GTK by default).

Reply:
I think I see what is tripping me up here. What seems logical to me would be to:

- Rename `requires_gtk` to something like `gtk_default_enabled`, default `True` for BSD/Linux which sets `enable_gtk` to `True`, allows disabling via `disable_gtk` (already done aside from renaming)
- Add `gtk_supported` internal var, `True` for BSD/Linux/mingw, allows enabling/disabling on these systems
- Update help output at bottom to display `note_unsupported` where `gtk_supported` is `False` (I'd also move this item to the end of the list, after libdovi)

Hope that makes some sense.

Reply:
Something in the section `# Only allow these features on supported platforms` would probably make the most sense for gating this.

Reply:
@bradleysepos
> Add `gtk_supported` internal var, `True` for BSD/Linux/mingw, allow enabling/disabling only on these systems

I'm not sure if this is necessary. The GTK-specific code should be portable enough to compile on any operating system. If someone wants to build HandBrake on an unsupported OS, it'll be easier if they can just enable the UI with a flag, rather than have to edit the script to add the OS manually.

Reply:
That's fine

Reply:
I guess my concern is if we're going to add a line to the bottom output showing GTK GUI is enabled/disabled, would hate for someone to think it's supported on their obscure platform because they see other instances of "not supported on target platform". So it's a consistency thing, perhaps. But not a hill I'm going to die on, either.

Reply:
Maybe the message is a little misleading? IMO, "not supported on target platform" sounds like a feature that is available but not officially tested or supported by the developers. If it was something like "not available on target platform" instead, it would more accurately reflect that the feature cannot be enabled on that platform.

Reply:
GTK GUI on systems other than BSD/Linux/mingw is "not supported" by that definition, so my previous suggestion still makes sense (to me at least). To date we have used the term to mean don't try it unless you are an engineer and want to tinker under the hood, so I guess I'm less enthused about changing the language here.

In the past, we've tried to encourage and allow only known good configurations, mostly because we're a small team and when things don't work, we get more support requests than we would like. Perhaps not ideal, but practical at times, and generally keeps everyday people from doing silly things.

Reply:
In the end I thought it would be best to add the code to disable on other platforms for now, just to be on the safe side. I added `_supported` variables for each of the platform-specific features in order to simplify the script by not repeating the same definitions in three different places, which also makes it easier for anyone interested to enable support for another platform.

Reply:
Seems reasonable, thanks.